### PR TITLE
Cleanup Block Diagonal Tests

### LINKED ISSUE #156: BlkDiagMatrix Unit Tests
The unit testing code has a lot of explicit arrays that can be reduced to controlled generative code.

I took a first pass at doing that in this commit, just to see what it would be like:
https://github.com/garrettwrong/ASPIRE-Python/commit/103c6cba33c6379051f98201f4cec403426103aa

Because the parent merge request #144 is on the large side already, creating this issue for future improvement.  It probably fits naturally with other unit test related issues.

### PR DESCRIPTION
**Pull request by garrettwrong**
Mainly convert from hard coded to generated arrays and some other cleanup.

Closes #156 .

Has code snippets relating to #277 .

### CODE REVIEW

**Comment by garrettwrong** [comment]
I accidentally added extra file, rebased, sorry about the extra noise.

File: tests/test_BlkDiagMatrix.py [hunk 1]
```diff
@@ -2,89 +2,44 @@
...
-        self.blk_partition = [(4, 4), (3, 3), (3, 3), (3, 3), (3, 3), (2, 2),
-                              (2, 2), (2, 2), (2, 2), (2, 2), (2, 2), (1, 1),
-                              (1, 1), (1, 1), (1, 1), (1, 1), (1, 1)]
+
+        self.num_blks = 10
+
+        self.blk_partition = [(i, i) for i in range(self.num_blks, 0, -1)]
+
+        n = np.sum(np.prod(np.array(self.blk_partition), axis=1))
+        self.Flat = np.arange(n)
```
Comment (janden) [inline tests/test_BlkDiagMatrix.py]: Lowercase for variable names (except matrices `A` and `B` below).

File: tests/test_BlkDiagMatrix.py [hunk 2]
```diff
@@ -332,119 +108,33 @@ def testBlkDiagMatrixSub(self):
...
+
+        # Check ndim 1 case
+        c = self.blk_a.apply(coeffm[:, 0])
+        self.allallfunc(c, res[:, 0])
+
+        # Check ndim 2 case
+        d = self.blk_a.apply(coeffm)
+        self.allallfunc(res, d)
+
+        e = np.empty((m, k))
```
Comment (janden) [inline tests/test_BlkDiagMatrix.py:143]: This one's a little tricky. Needs a short comment.

**Review by janden — CHANGES_REQUESTED** [review]

**Comment by garrettwrong** [comment]
Sure, I'll make these changes this morning.  Actually I wanted to ask you if you prefer the unit tests to follow a different approach.  Instead of manually making lists, I can expand the input and results blk diags to full matrices then use numpy operations, comparing the full matrix.  I wanted to clean things up this way first just to get my head back in it anyway, but if you would rather the tests go the np route I don't think it is much more work.

File: tests/test_BlkDiagMatrix.py (hunk 2 above)
Comment (garrettwrong) [reply tests/test_BlkDiagMatrix.py:143]: Added comment, and removed the addition (wasn't adding anything to the test).

**Comment by janden** [comment]
> Actually I wanted to ask you if you prefer the unit tests to follow a different approach. Instead of manually making lists, I can expand the input and results blk diags to full matrices then use numpy operations, comparing the full matrix.

There's merits to both ways, I think. Right now, I think this is fine, but it would be good to add a separate test to make sure that the full matrix representation is the correct one (say after a series of operations that would cover most of the functionality).

**Comment by garrettwrong** [comment]
> > Actually I wanted to ask you if you prefer the unit tests to follow a different approach. Instead of manually making lists, I can expand the input and results blk diags to full matrices then use numpy operations, comparing the full matrix.
> 
> There's merits to both ways, I think. Right now, I think this is fine, but it would be good to add a separate test to make sure that the full matrix representation is the correct one (say after a series of operations that would cover most of the functionality).

Cool, I added a test for the `dense()` method and also a sequence of arithmetic operations. There was a minor bug in `dense()` so it is good I went through again. Thanks.

File: tests/test_BlkDiagMatrix.py (hunk 2 above)
Comment (janden) [reply tests/test_BlkDiagMatrix.py:143]: Ah I see…

**Review by janden — APPROVED** [review]

**Pull request merged**